resources/js/components/dashboard/components/IndividualSponsor.tsx

```diff
@@ -0,0 +1,351 @@
+            sponsor_details: ["social-media","prizes","workshop","portal-info","demo-details"]
+        }).then(res => {
+            const status = res.status;
+            var portalInfo = undefined;
+            if(status >= 200 && status < 300) {
+                const data = res.data;
+                if(data && "success" in data && data["success"]) {
+                    const details = data["details"];
+                    if(Array.isArray(details)) {
+                        const detailState = details.reduce((result,r)=>{
+                            if (r.type==="portal-info"){
+                                const pInfo: {
+                                    data: {[varName: string]:string},
+                                    files: IAssetInformation[]
+                                } = JSON.parse(r["payload"]);
+                                portalInfo = pInfo;
+                            } else{
+                                const info: {
+                                    title: string,
+                                    description: string,
+                                    files: IAssetInformation[]
+                                } = JSON.parse(r["payload"]);
+                                // I'm pretty sure there's a nicer way to do this but 
+                                // I didn't have time to look it up. Spread operator?
+                                const rInfo = {
+                                    mainType: r.type,
+                                    title: info.title,
+                                    description: info.description,
+                                    files: info.files,
+                                }
+                                result.push(rInfo);
+                            }
+                            return result;
+                        },[])
+                        this.setState({ resources: detailState, portalInfo: portalInfo, loadingDefinitions: false });
+                        return;
+                    }
+                }
+            } else {
+                toast.error("Failed to load more information");
+                this.setState({ loadingDefinitions: true });
+            }
+
+            this.setState({ loadingDefinitions: false });
+        });
+    }
+
+    private renderResourceCard(data) {
+        // Title, description, files
+        return (
+            <Layout.Section oneThird>
+            <Card title={this.capitalizeFirstLetter(data.mainType)} sectioned>
+                {
+                    (!data || data.files.length === 0) ? <></> : <Image
+                    // source = {logoUrl}
+                    source={data.files[0].url}
+                    alt={data.files[0].name}
+                    style={{maxWidth:"100%",maxHeight:"100%"}}
+                />
+                }
+
+                <DisplayText>{data.title ? this.capitalizeFirstLetter(data.title) : "A resource!" }</DisplayText>
```

Comment:
Add a more descriptive note: `An untitled resource` and remove the exclamation mark